### PR TITLE
Add more tests for pkg/chrootarchive

### LINKED ISSUE #11601: pkg/chrootarchive - Test coverage
**Problem**
 
pkg/chrootarchive - Test coverage
 
**Solution**
 
The `pkg/chrootarchive` package is an essential piece of Docker security, yet the package test coverage could be improved (it's 40% at the time of filing this issue).
 
Improve test coverage on the `pkg/chrootarchive` package. You can use the following commands to measure progress:
 
    cd pkg/chrootarchive
    go test -cover -coverprofile /dev/stderr 2>&1 >/dev/null | go tool cover -func /dev/stdin

### PR DESCRIPTION
**Pull request by yestin**
Fixes #11601

Signed-off-by: Yestin Sun sunyi0804@gmail.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @unclejack

File: pkg/chrootarchive/archive_test.go [hunk 1]
```diff
@@ -45,6 +48,148 @@ func TestChrootTarUntar(t *testing.T) {
...
+	defer os.RemoveAll(tmpdir)
+	src := filepath.Join(tmpdir, "src")
+	if err := os.MkdirAll(src, 0700); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := prepareSourceDirectory(10, src, true); err != nil {
+		t.Fatal(err)
+	}
+	dest := filepath.Join(tmpdir, "dest")
+	if err := TarUntar(src, dest); err != nil {
```
Comment (LK4D4) [inline pkg/chrootarchive/archive_test.go]: Shouldn't we check somehow correctness of dest? maybe sha1 of src and dest?

Comment (yestin) [inline pkg/chrootarchive/archive_test.go]: Sound reasonable. I will create another PR with checking between src and dest.

**Comment by crosbymichael** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**